File: oop/lab3/java/Pentagon.java
public class Pentagon implements Figure {
    private double side;

    public Pentagon(double s) throws IllegalArgumentException {
        if (s <= 0) {
            throw new IllegalArgumentException("Side length must be positive.");
        }
        this.side = s;
    }

    @Override
    public double area() {
        return (5 * side * side) / (4 * Math.tan(Math.PI / 5));
    }

    @Override
    public double perimeter() {
        return 5 * side;
    }

    @Override
    public String name() {
        return "Pentagon";
    }
}

File: 2024_semester_2/programming-course/lab3/cpp/src/pentagon.cpp
#include <stdexcept>

#include "pentagon.hpp"

Pentagon::Pentagon(double s)
{
    if (s <= 0)
    {
        throw std::invalid_argument("Side length must be greater than 0.");
    }
    this->side = s;
}

double Pentagon::area() const noexcept(true)
{
    return 0.25 * sqrt(5 * (5 + 2 * sqrt(5))) * this->side * this->side;
}

double Pentagon::perimeter() const noexcept(true)
{
    return 5 * this->side;
}

const char *Pentagon::name() const noexcept(true)
{
    return "Pentagon";
}
